### PR TITLE
Fix CUDAToolkit_ROOT

### PR DESCRIPTION
**Pull request by jchen351**
This pull request primarily involves changes to the CUDA toolkit configuration in the `win-gpu-x64-build.yml` GitHub workflow and the `CMakeLists.txt` file. The changes ensure that the CUDA toolkit is correctly downloaded and its location is properly defined for subsequent use in the build process. 

Key changes include:

* [`.github/workflows/win-gpu-x64-build.yml`](diffhunk://#diff-0989dea22643962b5f5d6f36719798b1975f96bf0537ff0053c57c5e9a774e95R14): Added `CUDAToolkit_ROOT` environment variable to store the location of the CUDA toolkit.
* [`.github/workflows/win-gpu-x64-build.yml`](diffhunk://#diff-0989dea22643962b5f5d6f36719798b1975f96bf0537ff0053c57c5e9a774e95L28-R35): Modified the `Download cuda` job to download the CUDA toolkit to the workspace directory and added a new `Verify Cuda SDK` job to print the CUDA toolkit location for verification.
* [`.github/workflows/win-gpu-x64-build.yml`](diffhunk://#diff-0989dea22643962b5f5d6f36719798b1975f96bf0537ff0053c57c5e9a774e95L50-R57): Modified the `cmake` command in the `build` job to include the `CUDAToolkit_ROOT` environment variable, ensuring that the build process correctly locates the CUDA toolkit.
* [`CMakeLists.txt`](diffhunk://#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aL13-R17): Replaced the `check_language(CUDA)` command with `FindCUDAToolkit` to find the CUDA toolkit during the configuration process, and added a status message when CUDA is not found.

### CODE REVIEW

**Comment by jchen351** [comment]
This pull request primarily focuses on changes to the CUDA toolkit setup and usage in the GitHub Actions workflow and the CMake build system. The most significant changes include setting the `CUDAToolkit_ROOT` in the GitHub Actions environment, modifying the download and verification of the CUDA SDK in the workflow, updating the CMake command to use the `CUDAToolkit_ROOT` environment variable, and replacing the CUDA check in `CMakeLists.txt` to use `FindCUDAToolkit`.

Changes in GitHub Actions workflow:

* [`.github/workflows/win-gpu-x64-build.yml`](diffhunk://#diff-0989dea22643962b5f5d6f36719798b1975f96bf0537ff0053c57c5e9a774e95R14): Set `CUDAToolkit_ROOT` in the environment to point to the downloaded CUDA SDK.
* [`.github/workflows/win-gpu-x64-build.yml`](diffhunk://#diff-0989dea22643962b5f5d6f36719798b1975f96bf0537ff0053c57c5e9a774e95L28-R35): Modified the download of the CUDA SDK to store it in the GitHub workspace and added a step to verify the CUDA SDK.
* [`.github/workflows/win-gpu-x64-build.yml`](diffhunk://#diff-0989dea22643962b5f5d6f36719798b1975f96bf0537ff0053c57c5e9a774e95L50-R57): Updated the CMake command in the build step to use the `CUDAToolkit_ROOT` environment variable.

Changes in CMake build system:

* [`CMakeLists.txt`](diffhunk://#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aL13-R17): Replaced the check for CUDA support to use `FindCUDAToolkit` instead of `CheckLanguage`, and added a status message if CUDA is not found.

**Review by snnn — CHANGES_REQUESTED** [review]
Do not "Replaced the check_language(CUDA) ".  See the link I provided above.

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,9 +11,11 @@ set(USE_ORT_EXT 0) # "Build with Onnxruntime Extensions tokenizer support"
 
 # Checking if CUDA is supported
 include(CheckLanguage)
-check_language(CUDA)
-if (CMAKE_CUDA_COMPILER)
+find_package(CUDAToolkit)
```
[the remark this answers is not in the document]
Comment (jchen351) [reply CMakeLists.txt]: You can use [FindCUDAToolkit](https://cmake.org/cmake/help/git-stage/module/FindCUDAToolkit.html) to find a variety of useful targets and variables even without enabling the CUDA language.

[the remark this answers is not in the document]
Comment (jchen351) [reply CMakeLists.txt]: That is from the link you sent me.

**Pull request merged**